scripts/docgen.lua

```diff
@@ -161,6 +161,13 @@ local function make_lsp_sections()
                   make_section(0, '\n\n', sorted_map_table(default_settings.properties, function(k, v)
                     local function tick(s) return string.format("`%s`", s) end
                     local function bold(s) return string.format("**%s**", s) end
+
+                    -- https://github.github.com/gfm/#backslash-escapes
+                    local function excape_markdown_punctuations(str)
+                      local pattern = "\\(\\*\\|\\.\\|?\\|!\\|\"\\|#\\|\\$\\|%\\|'\\|(\\|)\\|,\\|-\\|\\/\\|:\\|;\\|<\\|=\\|>\\|@\\|\\[\\|\\\\\\|\\]\\|\\^\\|_\\|`\\|{\\|\\\\|\\|}\\)"
+                      return fn.substitute(str, pattern, "\\\\\\0", "g")
```

Comment:
lua raw string syntax  `[=[ ... ]=]` avoid the need for some escaping.

    local pattern = [=[\(\*\|\.\|?\|!\|...]=]
    return fn.substitute(str, pattern, [[\\\0]], "g")